src/methods/join.js

```diff
@@ -151,6 +151,22 @@ export class JoinImplementation {
     pnc: PossiblyNormalCompletion,
     c: PossiblyNormalCompletion
   ): PossiblyNormalCompletion {
+    //merge the two pathConditions
+    let mergedPath = [];
+    mergedPath.concat(pnc.path);
+    mergedPath.concat(c.path);
+    let savedPathConditions = []
```

Comment:
Because they should always be equal right?

Reply:
I don't think so. The second completion starts from a fork point inside the normal path of the first completion and presumably has its merged path as its saved path. The result, however, is a single completion with a single normal path, starting at the fork point that led to pnc, so the saved path should be the one that applied to pnc, since that is the only path that all of the threads collected in the result will have in common.